### DISCUSSION TITLE
Update data in rendered layer

### DISCUSSION THREAD

**Discussion by hypocritical20171014** [visgl/deck.gl#6855] [Q&A] [answered]
How  to complete the real-time update of data without using the DataFilter plugin, for example, I need to update my data in the already rendered layer

**Comment by Pessimistress** [comment, marked as answer]
Just create a new layer instance with the updated `data` value. See [rendering layers](https://deck.gl/docs/developer-guide/using-layers#rendering-layers)

**Reply by hypocritical20171014** [reply]
I see,but another question, if create a new layer with old layer's id, Will the layer properties of the old layer, such as color, etc., also be put into the new layer?
Do I need to create a new layer with all the properties of the old layer, but with different data?

**Reply by padawannn** [reply]
Yes, you need to create a new layer with all the properties of the old layer

**Reply by hypocritical20171014** [reply]
ah..thats doesn't sound easy，How can I get all the parameters of a layer that has been rendered?

**Reply by padawannn** [reply]
For example, if you code is something like this
```
function App({myData, viewState}) {
  const layer = new ScatterplotLayer({
    id: 'scatterplot-layer',
    data: myData,
    pickable: true,
    opacity: 0.8,
    stroked: true,
    filled: true,
    radiusScale: 6,
    radiusMinPixels: 1,
    radiusMaxPixels: 100,
    lineWidthMinPixels: 1,
    getPosition: d => d.coordinates,
    getRadius: d => Math.sqrt(d.exits),
    getFillColor: d => [255, 140, 0],
    getLineColor: d => [0, 0, 0]
  });

  return <DeckGL viewState={viewState} layers={[layer]}/>;
}
```
You only need to change your `myData` parameter and the layer automatically will be refreshed with the new data

**Reply by hypocritical20171014** [reply]
ok, this looks like a good way, maybe I should try to modify my logic code, thank you, it helps me a lot.
